### DISCUSSION TITLE
WifeGuard (wg-easy) with Telegram push notification

### DISCUSSION THREAD

**Discussion by kircanov** [wg-easy/wg-easy#1248] [Q&A]
i am looking for some upgrade to send notification to Telegram when user connected / disconnected from vpn.

**Comment by eandersons** [comment]
This would not be an easy task. As I see it, the solution would consist of two parts:

- a monitor that would periodically execute `wg show` (actually WireGuard Easy already does something similar to display connection information in the user interface: https://github.com/wg-easy/wg-easy/blob/babb9983aa3b3f212cc3d779929d424abc57cb4d/src/lib/WireGuard.js#L139-L189 https://github.com/wg-easy/wg-easy/blob/babb9983aa3b3f212cc3d779929d424abc57cb4d/src/www/js/app.js#L169-L232 https://github.com/wg-easy/wg-easy/blob/babb9983aa3b3f212cc3d779929d424abc57cb4d/src/www/js/app.js#L365-L369, but this only works when the user interface is opened in a browser) and parse its output to compare peer's handshake against the previous one (which would have to be stored somewhere) to determine if a client has connected after some period of inactivity or if they stopped using WireGuard Easy connection;
  when such events based on predefined rules would be detected, then the necessary information would be compiled and passed to the next part;
- Telegram sender - a simple application/module that composes and sends a message based on the received data to the configured endpoint;
  for this part a ready to use solution should already be available for any popular programming language.

After a quick look I did not manage to find a WireGuard monitoring solution that would emit events based on peers activity, which then could be listened to and used to compose and send a Telegram message. The closest I could find was [WireGuard module for Netdata](https://learn.netdata.cloud/docs/collecting-metrics/vpns/wireguard), but that would require a Netdata instance and as stated in modules page, it is not working with containerized WireGuard, and even if it would, then a solution to process the gathered data for sending it to Telegram would have to be found or developed.

**Reply by offizium-berndstorath** [reply]
Sending to telegram. Or creating a generic interface and creating a simple plugin like system would be doable.
Netdata just connects every second to the instance and gets the metrics. Something similar could be done with a new container (only for cronjob) or a worker thread

**Comment by johndoe3535** [comment]
Found a script that seems to do this https://github.com/alfiosalanitri/wireguard-client-connection-notification

**Reply by eandersons** [reply]
Just to keep in mind, depending on crontab implementation this script would be executed not more than once every minute. This would be a problem if more timely notifications are desired.

**Reply by offizium-berndstorath** [reply]
100% agree. But I think this would be out of scope. Checking so often just adds way to much load. 1m is a good interval imo. Using this project as a critical high precision solution is out of scope

**Comment by kircanov** [comment]
Thank you all, I somehow managed to solve it using dcron and curl where I did it on the container and used the https://github.com/alfiosalanitri/wireguard-client-connection-notification script for notification. It works quite well, I just need to fix the code to take Name from wg0.json instead of PublicKey as UserID.
